include/deal2lkit/parsed_error_estimator.h

```diff
@@ -0,0 +1,101 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2015 by the deal2lkit authors
+//
+//    This file is part of the deal2lkit library.
+//
+//    The deal2lkit library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE at
+//    the top level of the deal2lkit distribution.
+//
+//-----------------------------------------------------------
+
+#ifndef _d2k_parsed_solver_h
+#define _d2k_parsed_solver_h
+
+#include <deal.II/base/numbers.h>
+
+#include <deal2lkit/parameter_acceptor.h>
+#include <deal.II/numerics/error_estimator.h>
+#include <deal.II/base/function.h>
+#include <deal.II/dofs/function_map.h>
+#include <deal.II/fe/fe_tools.h>
+#include <deal.II/lac/vector.h>
+
+using namespace dealii;
+using namespace deal2lkit;
+
+D2K_NAMESPACE_OPEN
+
+/**
+ * TODO:
+ */
+template <int dim, int spacedim=dim, int n_components=dim>
+class ParsedErrorEstimator : public ParameterAcceptor
+{
+public:
+  /**
+   * TODO:
+   */
+  ParsedErrorEstimator( const std::string &name,
+                        const std::string &default_name,
+                        const std::string &_estimator_strategy =
+                          "cell_diameter_over_24");
+
+  /**
+   * TODO:
+   */
+  virtual void declare_parameters(ParameterHandler &prm);
+
+  /**
+   * TODO:
+   */
+  virtual void parse_parameters_call_back();
+
+  /**
+   * TODO:
+   */
+  template<typename InputVector>
+  void compute_estimator(
```

Comment:
Sbagliato. L'error estimator è un oggetto grosso. O implementi un move constructor per l'error estimator, oppure il codice sopra lavora dentro la funzione su una copia del vettore, e poi ne **crea un'altra copia** in uscita. L'utente deve sapere qual e' la dimensione di error estimator. L'uso dovrebbe essere questo:

~~~
ParsedErrorEstimator<dim, spacedim, components> pe;

...

Vector<float> error_per_cell(tria.n_active_cells());
pe.compute_estimator(dh, solution, error_per_cell)
pgg.mark_cells(error_per_cell, tria);
tria.execute_coarsening_and_refinement();
~~~